### PR TITLE
trial to add the button -delete me from event (for members) in the ev…

### PR DESCRIPTION
**Pull request by kasiaekiert**
…ent/show

### CODE REVIEW

File: app/controllers/members_controller.rb [hunk 1]
```diff
@@ -4,5 +4,14 @@ def create
     event = Event.find(params[:event_id])
     event.members << current_user
     redirect_back(fallback_location: root_path)
-  end 
+  end
+
+  def destroy
+    event = Event.find(params[:event_id])
```
Comment (womanonrails) [inline app/controllers/members_controller.rb]: Right now you have in two methods the same code: `event = Event.find(params[:event_id])` try to extract it to separate method. You have similar thing in events and places controller.

Comment (womanonrails) [reply app/controllers/members_controller.rb]: Still valid.

Comment (kasiaekiert) [reply app/controllers/members_controller.rb]: I have written code for it... hmm - now I see it on branch remove_event

```
  before_action :set_event, only: [:show, :edit, :update, :destroy]
```
```
def set_event
      @event = Event.find(params[:id])
    end 
```

Do you think about it?

Comment (womanonrails) [reply app/controllers/members_controller.rb]: Yes this is what I meant, but in this controller you don't have action `:show, :edit` or `:update`. If I remember correctly you have `:create` and `:destroy`

Comment (kasiaekiert) [reply app/controllers/members_controller.rb]: I have update it on branch button_remove_me

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class EventsController < ApplicationController
-  before_action :set_event, only: [:show, :edit, :update, :destroy]
```
Comment (womanonrails) [inline app/controllers/events_controller.rb]: From what I know you have this all actions in event controller and it was OK here. Did you test it in the browser after change? We discussed to remove this code from member controller.

File: app/controllers/members_controller.rb (hunk 1 above)
Comment (womanonrails) [reply app/controllers/members_controller.rb]: Why not here? All changes with delete button should be one this branch or you create new one?

File: app/views/events/index.html.erb [hunk 3]
```diff
@@ -24,13 +24,14 @@
...
             <% if event.has_member?(current_user) %>
               <%= t('.you_are_member') %>
+              <%= link_to t('links.destroy_me'), event_members_path(event), method: :destroy %>
             <% else %>
               <%= link_to t('links.add_me'), event_members_path(event), method: :post %>
             <% end %>
           </td>
           <td>
             <% if can? :manage, event %>
-              <%= link_to t('links.destroy'), event, method: :delete, data: { confirm: 'Are you sure?' } %>
```
Comment (womanonrails) [inline app/views/events/index.html.erb]: There is no `destroy` method when you do HTTP request you can choose between `GET, POST, PUT, PATCH, DELETE` so previous version was OK.

File: app/controllers/members_controller.rb [hunk 4]
```diff
@@ -1,8 +1,17 @@
...
     event = Event.find(params[:event_id])
     event.members << current_user
     redirect_back(fallback_location: root_path)
-  end 
-end
+  end
+
+  def destroy
+    event = Event.find(params[:event_id])
+    event.members = current_user
```
Comment (womanonrails) [inline app/controllers/members_controller.rb]: You don't what to assign `current_user` to members. You want to remove it from members. And to do that you need to check if you have permission to do that. Like we discusses what should happen when user is member or owner of event.

**Pull request merged**